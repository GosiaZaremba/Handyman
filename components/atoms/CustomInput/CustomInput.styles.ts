import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  input: {fontFamily: 'OpenSans-Bold'},
  inputContainer: {
    width: '100%',
  },
});
