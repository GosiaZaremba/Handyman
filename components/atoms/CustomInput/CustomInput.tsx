import React, {forwardRef, useImperativeHandle, useRef, useState} from 'react';
import {TextInput, View} from 'react-native';
import styles from './CustomInput.styles';

export type CustomInputReference = {
  getValue: () => string;
  focus: () => void;
  isFocused: () => boolean | undefined;
};

const CustomInputWithReference: React.ForwardRefRenderFunction<
  CustomInputReference
> = ({}, ref) => {
  const [value, setValue] = useState<string>('');

  const inputReference = useRef<TextInput>(null);

  const onChangeText = (val: string) => {
    setValue(val);
  };

  useImperativeHandle(ref, () => ({
    getValue: () => {
      return value;
    },
    focus: () => {
      inputReference.current?.focus();
    },
    isFocused: () => {
      return inputReference.current?.isFocused();
    },
  }));

  return (
    <View style={styles.inputContainer}>
      <TextInput
        onChangeText={onChangeText}
        placeholder={' '}
        secureTextEntry={false}
        value={value}
        style={styles.input}
        ref={inputReference}
      />
    </View>
  );
};

export const CustomInput = forwardRef(CustomInputWithReference);
