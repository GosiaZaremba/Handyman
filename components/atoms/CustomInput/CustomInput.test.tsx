import React, {createRef} from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import {CustomInput, CustomInputReference} from './CustomInput';

describe('Atom CustomInput', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('Component Snapshot', () => {
    const tree = renderer
      .create(<CustomInput testID="atom-customInput" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure reference is passed', () => {
    const inputRef = createRef<CustomInputReference>();
    const {getByTestId} = render(
      <CustomInput testID="atom-customInput" ref={inputRef} />,
    );
    const customInput = getByTestId('atom-customInput');
    fireEvent.changeText(customInput, 'input text');
    expect(inputRef.current?.getValue()).toBe('input text');
    expect(inputRef.current?.focus()).toBeUndefined();
  });
});
