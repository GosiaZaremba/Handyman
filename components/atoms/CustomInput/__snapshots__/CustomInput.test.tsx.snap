// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom CustomInput Component Snapshot 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder=" "
    secureTextEntry={false}
    style={
      Object {
        "fontFamily": "OpenSans-Bold",
      }
    }
    testID="atom-customInput"
    value=""
  />
</View>
`;
